### ISSUE TITLE
Crop Pest and Disease Detection

### ISSUE THREAD

**Issue by abhisheks008** [abhisheks008/DL-Simplified#460]
### Deep Learning Simplified Repository (Proposing new issue)
:red_circle: **Project Title** : Crop Pest and Disease Detection</br>
:red_circle: **Aim** : The aim of this project is to make a deep learning project solving the given problem statement.</br>
:red_circle: **Dataset** : https://www.kaggle.com/datasets/nirmalsankalana/crop-pest-and-disease-detection</br>
:red_circle: **Approach** : Try to use 3-4 algorithms to implement the models and compare all the algorithms to find out the best fitted algorithm for the model by checking the accuracy scores. Also do not forget to do a exploratory data analysis before creating any model.

**********************************************************************************
### 📍 **Follow the Guidelines to Contribute in the Project :**
* You need to create a separate folder named as the Project Title.
* Inside that folder, there will be four main components.
   * Images - To store the required images.
   * Dataset - To store the dataset or, information/source about the dataset.
   * Model - To store the machine learning model you've created using the dataset.
   * `requirements.txt` - This file will contain the required packages/libraries to run the project in other machines.
* Inside the `Model` folder, the `README.md` file must be filled up properly, with proper visualizations and conclusions.

***********************************************************************
:red_circle::yellow_circle: **Points to Note :**

- The issues will be assigned on a first come first serve basis, 1 Issue == 1 PR.
- "Issue Title" and "PR Title should be the same. Include issue number along with it.
- Follow Contributing Guidelines & Code of Conduct before start Contributing.

***********************************************************************
:white_check_mark: **To be Mentioned while taking the issue :**
- Full name : 
- GitHub Profile Link : 
- Email ID :
- Participant ID (if applicable):
- Approach for this Project :
- What is your participant role? (Mention the Open Source program)

*************************************************************
Happy Contributing 🚀 

All the best. Enjoy your open source journey ahead. 😎

**Comment by tushtithakur** [comment]
Hi , I'm excited to contribute to this project. Could you please assign me? Looking forward to getting started! @abhisheks008

Full name : Tushti Thakur
GitHub Profile Link : https://github.com/tushtithakur
Email ID : [tushtithakur1234@gmail.com](mailto:tushtithakur1234@gmail.com)
Approach for this Project : Implement different deep learning algorithms using the dataset, evaluate it and compare performance.
What is your participant role? GSSoC 2024

**Comment by shaziaakhan** [comment]
Full name : Shazia Khan
GitHub Profile Link : https://github.com/shaziaakhan
Email ID : [shazia.takhan@gmail.com]
Approach for this Project : Implement ML algorithms and analyze the accuracy
What is your participant role?: GSSoC '24 Contributor

**Comment by abhisheks008** [comment]
@shaziaakhan need some clarity on the planned deep learning models for this project by you.

**Comment by Medhaj02** [comment]
Full name: Medhaj Sharma
Github profile Link: https://github.com/Medhaj02
Email Id: medhaj.sharma08@gmail.com
Approach for this project: Since the main goal of this project is to detect whether the crop yield is affected by crop pest and if there is some kind of disease affecting the crop so, I would use 
1.CNN(Convolutional Neural Network0 as there could be chances that i have to perform image processing to check for the quality of yield.
2. Long Short Term Memory Network to learn about what chemicals were used and how much care was given to the crop as this algo is useful for time-series prediction.
3. Multilayer Perceptrons to compute which crops gave positive results and the ones that were detected to have crop pest and disease
4. Self organizing maps for visualizing data.
What is your participant role? GSSoC 2024 Contributor

**Comment by shaziaakhan** [comment]
@abhisheks008 I would be using a CNN model

**Comment by abhisheks008** [comment]
Hi @shaziaakhan while commenting in this issues, you need to be specific and brief about your approach. Otherwise your comments will not get that much traction, even though you have expertise in the issue. Hope this will help you in future.

Issue assigned to @Medhaj02. You can start working on it as your approach looks good to me for this issue.

**Comment by Medhaj02** [comment]
thank you @abhisheks008 for giving me this opportunity. Will start working on this one

**Comment by jyothi200217** [comment]
Sir can you please help to do this project

**Comment by bhoomika0910** [comment]
Hi! I'm a GSSoC '26 contributor with hands-on 
experience in Python, Deep Learning, and Computer 
Vision (currently building a YOLOv8-based assistive 
system). I'd love to work on this project and submit 
a well-documented Jupyter Notebook with model 
training, evaluation, and results.

Could you please assign this to me? 🙏

**Comment by abhisheks008** [comment]
Hi @bhoomika0910 
- please suggest 4 models atleast
- Reason behind choosing the models?
- How will you judge the models?

**Comment by bhoomika0910** [comment]
Hi @abhisheks008! Thank you for the response 😊

Here are my proposed models:

1. CNN – Baseline model for image classification
2. ResNet-50 – Transfer learning, fine-tuned for 
crop disease detection
3. MobileNetV2 – Lightweight, good for deployment
4. VGG16 – Strong feature extractor for comparison

Why these models:
These cover simple to advanced architectures for 
meaningful accuracy vs efficiency comparison.

How I'll judge them:
- Accuracy & Validation Accuracy
- Precision, Recall, F1-Score
- Confusion Matrix
- Loss curves (Training vs Validation)

Dataset: PlantVillage dataset (Kaggle)

Please assign this to me, I'll submit a 
well-documented Jupyter Notebook! 🙏
